examples/nlp/summarization.py

```diff
@@ -0,0 +1,378 @@
+data will be preprocessed in one single command.
+"""
+
+tokenized_datasets = raw_datasets.map(preprocess_function, batched=True)
+
+"""
+## Defining the model
+"""
+
+"""
+Now we can download the pretrained model and fine-tune it. Since our task is
+sequence-to-sequence (both the input and output are text sequences), we use the
+`TFAutoModelForSeq2SeqLM` class from the 🤗 Transformers library. Like with the
+tokenizer, the `from_pretrained` method will download and cache the model for us.
+
+The `from_pretrained()` method expects the name of a model from the 🤗 Model Hub. As
+mentioned earlier, we will use the `t5-base` model checkpoint.
+"""
+
+from transformers import TFAutoModelForSeq2SeqLM, DataCollatorForSeq2Seq
+
+model = TFAutoModelForSeq2SeqLM.from_pretrained(MODEL_CHECKPOINT)
+
+"""
+For training Sequence to Sequence models, we need a special kind of data collator,
+which will not only pad the inputs to the maximum length in the batch, but also the
+labels. Thus, we use the `DataCollatorForSeq2Seq` provided by the 🤗 Transformers
+library on our dataset. The `return_tensors='tf'` ensures that we get `tf.Tensor`
+objects back.
+"""
+
+from transformers import DataCollatorForSeq2Seq
+
+data_collator = DataCollatorForSeq2Seq(tokenizer, model=model, return_tensors="tf")
+
+"""
+Next we define our training and testing sets with which we will train our model. Again, 🤗
+Datasets provides us with the `to_tf_dataset` method which will help us integrate our
+dataset with the `collator` defined above. The method expects certain parameters:
+
+- **columns**: the columns which will serve as our independant variables
+- **batch_size**: our batch size for training
+- **shuffle**: whether we want to shuffle our dataset
+- **collate_fn**: our collator function
+
+Additionally, we also define a relatively smaller `generation_dataset` to calculate
+`ROUGE` scores on the fly while training.
+"""
+
+train_dataset = tokenized_datasets["train"].to_tf_dataset(
+    batch_size=BATCH_SIZE,
+    columns=["input_ids", "attention_mask", "labels"],
+    shuffle=True,
+    collate_fn=data_collator,
+)
+test_dataset = tokenized_datasets["test"].to_tf_dataset(
+    batch_size=BATCH_SIZE,
+    columns=["input_ids", "attention_mask", "labels"],
+    shuffle=False,
+    collate_fn=data_collator,
+)
+generation_dataset = (
+    tokenized_datasets["test"]
+    .shuffle()
+    .select(list(range(200)))
+    .to_tf_dataset(
+        batch_size=BATCH_SIZE,
+        columns=["input_ids", "attention_mask", "labels"],
+        shuffle=False,
+        collate_fn=data_collator,
+    )
+)
+
+"""
+## Building and Compiling the the model
+"""
+
+"""
+Now we will define our optimizer and compile the model. The loss calculation is handled
+internally and so we need not worry about that!
+"""
+
+from transformers import AdamWeightDecay
+
+optimizer = AdamWeightDecay(learning_rate=LEARNING_RATE, weight_decay_rate=WEIGHT_DECAY)
```

Comment:
Q: does it work for you to use `keras.optimizers.experimental.AdamW`?

Reply:
Hi @fchollet ,

I just realized that I might need to revert this change. My model isn't training well after this, and the loss doesn't decrease ( I just changed the optimizer, nothing else). I will revert this change.